### ISSUE TITLE
Cannot read property 'type' of undefined

### ISSUE THREAD

**Issue by praveenaj** [mozfreddyb/eslint-plugin-scanjs-rules#8]
I'm getting undefined error in object_mozSystem.js rule.

```
Praveenas-MacBook-Pro:webchat praveenaj$ ./node_modules/.bin/eslint src/*
Cannot read property 'type' of undefined
TypeError: Cannot read property 'type' of undefined
    at EventEmitter.ObjectExpression (/Repo/node_modules/eslint-plugin-scanjs-rules/lib/rules/object_mozSy
stem.js:14:21)

    at emitOne (events.js:95:20)
    at EventEmitter.emit (events.js:182:7)
    at NodeEventGenerator.applySelector (/Repo/node_modules/eslint/lib/util/node-event-generator.js:265:26
)
    at NodeEventGenerator.applySelectors (/Repo/node_modules/eslint/lib/util/node-event-generator.js:294:2
2)
    at NodeEventGenerator.enterNode (/Repo/node_modules/eslint/lib/util/node-event-generator.js:308:14)
    at CodePathAnalyzer.enterNode (/Repo/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:
602:23)
    at CommentEventGenerator.enterNode (/Repo/node_modules/eslint/lib/util/comment-event-generator.js:98:2
3)
    at Traverser.enter (/Repo/node_modules/eslint/lib/eslint.js:929:36)
    at Traverser.__execute (/Repo/node_modules/estraverse/estraverse.js:397:31)
```

**Comment by mozfreddyb** [comment]
Thank you for reporting this bug, @praveenaj.
It's odd that eslint does not say in which scanned source line that error occurs.
Can you help me find the relevant line in your source code?

**Comment by jseppi** [comment]
I just encountered this problem as well. Here is a sample of code that will cause the error:

```js
const x = { a: 1, b: 2 };

function hi() { };

hi({...x});  // this line causes the `Cannot read property 'type' of undefined` exception
```

**Comment by mozfreddyb** [comment]
Fixed with the most recent commits, which also created a bubfix release 0.1.6.
Thanks for helping me find an offending code line, @jseppi!

**Comment by mozfreddyb** [comment]
NB, I have also just removed a truck load of tests that were used for proprietary DOM extensions from Firefox OS and released 0.1.7. The number of rules went down from 181 to 78, which hopefully speeds up things significantly

**Comment by jseppi** [comment]
Thanks!
